### DISCUSSION TITLE
Playlist.ini does not honor privacy settings under certain situations

### DISCUSSION THREAD

**Discussion by Raza-0007** [smplayer-dev/smplayer#898] [General]
@smplayer-dev,

There is an issue that I noticed with `playlist.ini`. Here is a detailed description.

Screenshots of my settings are attached. My playlist is built automatically when I play a file from a folder and the current playlist is saved when I close smplayer. This helps me to resume from where I left off.

Let’s suppose I have the following three folders

- Folder 1 with 10 items
- Folder 2 with 15 items
- Folder 3 with 20 items

Based on my settings, when I open a file from folder 1 and then close smplayer, 10 items are saved in `playlist.ini`. If I now open a file from folder 2 and close smplayer, my previous playlist entries are overwritten and now I have 15 items from folder 2 in `playlist.ini`. Playing a file from folder 3 again overwrites the `playlist.ini` with 20 items from folder 3.

Everything looks good and works as expected!

The issue occurs when you reverse the order of playing. If I open a file from folder 3 first, the `playlist.ini` gets populated with 20 items. Next opening a file from folder 2 should overwrite the previous playlist contents but it does not! Only the first 15 items are overwritten, and items 16-20 are from previous playlist. Opening folder 1 only overwrite the first 10 items leaving 11-15 from folder 2 and 16-20 from folder 3. 

Thus, if I open a folder with 300 items, those items stay in the `playlist.ini` until another folder with even larger list of items is played, creating a privacy nightmare, as `playlist.ini` will retain those files until they are overwritten.

Just wanted to add that these items are only left in `playlist.ini`, the playlist itself stops as expected after the last item in the current folder is played.

This is not an urgent issue but when you get some free time can you please fix this. 

Thanks,

~Raza

.

![Privacy](https://github.com/smplayer-dev/smplayer/assets/131807929/e783a460-c521-4c04-bfe6-e6fc0c967ac2)
![Playlist 1](https://github.com/smplayer-dev/smplayer/assets/131807929/aa7692ee-9b24-44e6-ab85-4344ab8017f1)
![Playlist 2](https://github.com/smplayer-dev/smplayer/assets/131807929/d678b875-cdf5-4a6f-82e1-cf2b9e0a07bf)
![smplayer](https://github.com/smplayer-dev/smplayer/assets/131807929/94c4ecdc-0d44-4c4d-b252-f0510ae8c368)

**Comment by smplayer-dev** [comment]
I think it's fixed in this commit: 403b57ec88cbad4df37ce36e1dc449e04cffeb8b

**Reply by Raza-0007** [reply]
Thanks, I will test it when you next post a pre-release build.

~Raza

**Reply by Raza-0007** [reply]
I tested it and can confirm that this issue has been fixed in the latest test release 23.12.0.10229

Thanks,

~Raza
